src/AppBundle/Controller/Admin/UserController.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace AppBundle\Controller\Admin;
+
+use AppBundle\Form\Model\UserWithPlainPassword;
+use AppBundle\Form\UserWithPlainPasswordType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use AppBundle\Entity\User;
+use AppBundle\Form\UserType;
+
+/**
+ * User controller.
+ * @Route("/user")
+ */
+class UserController extends Controller
+{
+
+    /**
+     * Lists all User entities.
+     *
+     * @Route("/", name="admin_user")
+     * @Method("GET")
+     * @Template()
+     */
+    public function indexAction()
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $entities = $em->getRepository('AppBundle:User')->findAll();
+
+        return [
+            'entities' => $entities,
+        ];
+    }
+
+    /**
+     * Creates a new User entity.
+     *
+     * @Route("/", name="admin_user_create")
+     * @Method("POST")
+     * @Template("AppBundle:Admin/User:new.html.twig")
+     */
+    public function createAction(Request $request)
+    {
+        $entity = new User();
+        $form   = $this->createCreateForm($entity);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $entity->setPassword(
+                $this->container->get('security.password_encoder')->encodePassword($entity, $entity->getPassword())
+            );
+            $em->persist($entity);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('admin_user'));
+        }
+
+        return [
+            'entity' => $entity,
+            'form'   => $form->createView(),
+        ];
+    }
+
+    /**
+     * Creates a form to create a User entity.
+     *
+     * @param User $entity The entity
+     *
+     * @return \Symfony\Component\Form\Form The form
+     */
+    private function createCreateForm(User $entity)
+    {
+        $form = $this->createForm(new UserType(), $entity, [
```

Comment:
Form should be created using name. It will allow us to use form extensions and dependencies inside form types